aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -36,7 +36,7 @@ void window_function_checks(
     const TensorOptions& options,
     int64_t window_length) {
   TORCH_CHECK(
-      options.layout() != kSparse,
```

Comment:
These changes don't look right. They're not implemented for `SparseCOO`, they're probably not implemented for `SparseGCS` either. A more correct fix would be to invert these layout tests so that we test for what layouts the factory *actually supports*. That would be a good preparatory PR to stack before your main GCS PR.

Reply:
I think simply getting GCS to work without touching COO for now would be a good idea for a start. So I've removed these changes relating to COO altogether.